attestation-agent/attester/src/cca/mod.rs

```diff
@@ -1,105 +1,56 @@
-// Copyright (c) 2023 Arm Ltd.
+// Copyright (c) 2023-2024 Arm Ltd.
 //
 // SPDX-License-Identifier: Apache-2.0
 //
 
+use super::tsm_report::*;
 use super::Attester;
 use anyhow::*;
-use base64::Engine;
-use nix::fcntl::{open, OFlag};
-use nix::sys::stat::Mode;
-use nix::unistd::close;
 use serde::{Deserialize, Serialize};
-use std::path::Path;
 
-const CCA_DEVICE_PATH: &str = "/dev/cca_attestation";
+const CCA_CHALLENGE_SIZE: usize = 64;
 
-// NOTE: The path might be different when the CCA feature is public available, will come back to update the actual path if needed.
 pub fn detect_platform() -> bool {
-    Path::new(CCA_DEVICE_PATH).exists()
+    #[cfg(target_arch = "aarch64")]
+    return TsmReportPath::new(TsmReportProvider::Cca).is_ok();
+    #[cfg(not(target_arch = "aarch64"))]
+    return false;
 }
 
 #[derive(Debug, Default)]
-pub struct CCAAttester {}
+pub struct CcaAttester {}
 
 #[derive(Serialize, Deserialize)]
 struct CcaEvidence {
     /// CCA token
     token: Vec<u8>,
 }
 
-#[allow(non_camel_case_types)]
-#[repr(C)]
-pub struct cca_ioctl_request {
-    challenge: [u8; 64],
-    token: [u8; 4096],
-    token_length: u64,
-}
-
-nix::ioctl_readwrite!(cca_attestation_request, b'A', 1, cca_ioctl_request);
-
 #[async_trait::async_trait]
-impl Attester for CCAAttester {
+impl Attester for CcaAttester {
     async fn get_evidence(&self, mut challenge: Vec<u8>) -> Result<String> {
-        challenge.resize(64, 0);
-        let token = attestation(challenge)?;
+        if challenge.len() > CCA_CHALLENGE_SIZE {
+            bail!("CCA Attester: Challenge size must be {CCA_CHALLENGE_SIZE} or less.");
+        }
+
+        challenge.resize(CCA_CHALLENGE_SIZE, 0);
```

Comment:
Does this `0` mean char `0` or `\0`? I assume you mean `\0` here

Reply:
it likely follows `tdx-attester` here and aims to fulfill "Up to 64 bytes of user specified binary data" for `inblob`.

Reply:
Right. Just to ensure the padding is `\0` rather than `0`, and it is.